File: electricDoor/esp12f_relay/persistance.py
import json
import ubinascii

WIFI_SSID = 'wifiSsid'
WIFI_PASSWORD = 'wifiPassword'
ACTIVE = 'active'
NAME = 'name'
ID = 'id'
HOME_ID = 'homeId'
MQTT_BROKER = 'mqttBrokerId'
MQTT_USER = 'mqttUser'
MQTT_PASSWORD = 'mqttPassword'

fileName = 'persistance.json'

def read():
    with open(fileName,'r') as f:
        value = f.read()
        f.close()
        return value

def write(content):
    with open(fileName,'w') as f:
        f.write(content)
        f.close()

def clear():
    write('{}')

def init():
    try:
       read()
    except:
        clear()

def get_json():
    try:
        value = read()
        return json.loads(value)
    except:
        print('could not read file')
        return {};

def get(key):
    try:
        return get_json()[key]
    except:
        print('cound not get for key: ', key)

def persist(key, value):
    response = get_json()
    response.update({key: value})
    write(json.dumps(response))

def getToken(initial=False):
    try:
        name = get(HOME_ID)
        if not initial:
            name+='/'+get(ID)
        return ubinascii.b2a_base64(f'{name}/user:keylock'.encode()).decode().strip()
    except Exception as e:
        print('Error:',e)
        return ''

init()

File: arduino/doorMotor/esp12f_1ch/Persistance.cpp
#include "Persistance.h"

void Persistance::init() {
  if (!LittleFS.begin()) {
    Serial.println("Failed to mount file system");
    return;
  }
  if (read().isEmpty()) {
    clearJson();
  };
}

String Persistance::read(const String& name) {
  File file = LittleFS.open(name, "r");
  if (!file) {
    Serial.println("Failed to open file for reading " + name);
    return "";
  }
  return file.readString();  // file is closed automatically when 'file' goes out of scope
}

void Persistance::write(const String& name, const String& text) {
  File file = LittleFS.open(name, "w");
  if (!file) {
    Serial.println("Failed to open file for writing " + name);
    return;
  }
  file.println(text);
  file.close();
}

String Persistance::read() {
  return read(psFileName);
}

void Persistance::write(const String& text) {
  write(psFileName, text);
}

void Persistance::clearJson() {
  write("{}");
}

DynamicJsonDocument Persistance::getJson() {
  DynamicJsonDocument doc(1024);
  DeserializationError error = deserializeJson(doc,read());
  if (error) {
    doc.clear();
  }
  return doc;
}

String Persistance::get(const String& key) {
  return getJson()[key].as<String>();
}

void Persistance::persist(const String& key, const String& value) {
  DynamicJsonDocument doc = getJson();
  doc[key] = value;
  String prettyOutput;
  serializeJsonPretty(doc, prettyOutput);
  write(prettyOutput);
}

String Persistance::getToken() {
  return base64::encode(get(HOME_ID) + "/" + get(ID) + "/user:keylock");
}
